### ISSUE TITLE
Test error always around 0% accuracy

### ISSUE THREAD

**Issue by hyenal** [yanbeic/semi-memory#2]
Hi and thanks for the great work!
I noticed after running the `train_svhn_semi.sh` script that test accuracy is always around zero percent.
The same problem happens again if I train with the supervised setting (this time prediction is always 5 for some reason).
In both case training reaches 98/99% accuracy after only 2 epochs which seems a bit strange.
I'm using tensorflow 1.9.0 and python 2.7.6

**Comment by yanbeic** [comment]
Hi,

The printed results at test time (precision@1) are not multiplied with 100. (test.py line 91)
The printed results during training (top 1 scores) are multiplied with 100. (semi_learner.py line 151)

So actually you need to multiple 100 if you are measuring as percentage (%).

**Comment by hyenal** [comment]
Hi, the accuracy I talked about was in percent for both.

**Comment by yanbeic** [comment]
Hi,

Could you paste the results here? 

Ideally, the printed results are something as follows during training/testing:
(1) training: 
epoch 1, L = 3.84, Ls = 1.63, Lm = 2.05, top_1 = 42.23, lr = 0.0200 (time_per_iter: 0.0623 s)
(2) testing:
2018-09-17 20:54:19.643729: precision @ 1 = 0.9607 recall @ 5 = 0.9921

As said before, top_1 = 42.23 means an accuracy of 42.23%. 
precision @ 1 = 0.9607 means an accuracy of 96.07%.

**Comment by hyenal** [comment]
testing: precision @ 1 = 0.0062 recall @ 5 = 0.4145

**Comment by hyenal** [comment]
These are the results obtained for the first three epochs on training
epoch 0, L = 3.76, Ls = 1.59, Lm = 2.02, top_1 = 44.25, lr = 0.0199 (time_per_iter: 0.3763 s)
epoch 1, L = 1.42, Ls = 0.36, Lm = 0.90, top_1 = 91.30, lr = 0.0198 (time_per_iter: 0.3668 s)
epoch 2, L = 0.82, Ls = 0.20, Lm = 0.46, top_1 = 96.93, lr = 0.0197 (time_per_iter: 0.3637 s)

**Comment by yanbeic** [comment]
The training acc looks fine as it reflects only a small batch of labelled data.
The testing acc is weird. 
We double-check and re-run this code which gives
`2019-01-23 23:45:18.043114: precision @ 1 = 0.9575 recall @ 5 = 0.9886 [26032 examples]`
(tested with tensorflow 1.5.0 and python 2.7.12.)

Nothing turns out to cost a weird result of 0%.

**Comment by hyenal** [comment]
Thanks for your quick reply. I spotted the issue but I havent found a good fix for it. I was solving the models and data on another machine so `results/` and `tfrecords/` were symlinked. It seems like it might cause an issue in this case. I think you can close the issue as it is not related to your code.

**Comment by hyenal** [comment]
Thanks for your quick reply. I spotted the issue but I havent found a good fix for it. I was solving the models and data on another machine so `results/` and `tfrecords/` were symlinked. It seems like it might cause an issue in this case. Im closing the issue since it is not related to your code.
